src/mod_pushoff_fcm.erl

```diff
@@ -226,25 +210,40 @@ restart_retry_timer(OldTimer) ->
 
-pending_element_to_json(Element) ->
-    case Element of 
-        {_MessageId, {_, _Payload, _Token, _DisableArgs}} ->
-            _Body = proplists:get_value(body, _Payload),
-            _From = proplists:get_value(from, _Payload),
-            PushMessage = {struct, [
-                                    {to,           _Token},
-                                    {priority,     <<"high">>},
-                                    {data,         {struct, [
-                                                             {body, _Body},
-                                                             {title, _From}
-                                                            ]}}%,
-                                    %{notification, {struct, [
-                                    %                         {body, _Body},
-                                    %                         {title, _From}
-                                    %                        ]}}
-                                   ]},
-            {iolist_to_binary(mochijson2:encode(PushMessage)), _DisableArgs};
-        _ ->
-            ?ERROR_MSG("no pattern for matching for pending list", []),
-            unknown
+pending_element_to_json({_MessageId, {_, _Payload, _Token, _DisableArgs}}) ->
+    _Body = proplists:get_value(body, _Payload),
+    _From = proplists:get_value(from, _Payload),
+    PushMessage = {struct, [
+                            {to,           _Token},
+                            {priority,     <<"high">>},
+                            {data,         {struct, [
+                                                     {body, _Body},
+                                                     {title, _From}
+                                                    ]}}%,
+                            %{notification, {struct, [
+                            %                         {body, _Body},
+                            %                         {title, _From}
+                            %                        ]}}
+                           ]},
+    {iolist_to_binary(mochijson2:encode(PushMessage)), _DisableArgs};
+
+pending_element_to_json(_) ->
+    unknown.
+
+parse_responce(ResponseBody, DisableArgs) ->
+    {struct, JsonData} = mochijson2:decode(ResponseBody),
+    case proplists:get_value(<<"success">>, JsonData) of
+        1 ->
+            ok;
+        0 ->
+            [{struct, ResultList}] = proplists:get_value(<<"results">>, JsonData),
+            case proplists:get_value(<<"error">>, ResultList) of
+                <<"NotRegistered">> ->
+                    ?ERROR_MSG("FCM error: invalid token, unregistered user", []),
+                    mod_pushoff:unregister_client(DisableArgs);
+                <<"InvalidRegistration">> ->
+                    ?ERROR_MSG("FCM error: invalid registration, unregistered user", []),
+                    mod_pushoff:unregister_client(DisableArgs);
```

Comment:
you called the function `parse_response` yet it is invoking `unregister_client` itself.
maybe make it return an atom like`unregister` instead and do effects at the call site?